### PR TITLE
Prevent Service Account annotation conflicts on OpenShift 4.16

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Bugfix

### Description

It looks like we have on OCP 4.16 a conflict in the reconciliation of Service Accounts.

OpenShift 4.16 will create an image pull secret, add it to the image pull secrets list, and attach an annotation `openshift.io/internal-registry-pull-secret-ref` to the Service Account with the name of the image pull secret. If Strimzi removes this annotation by patching the resource in the next reconciliation, OpenShift will create another Secret, add it to the image pull secret list and add back the annotation. This way, every reconciliation will for every Service Account make OCP generate a new Secret and in a little while, the namespace is full of these Secrets.

This PR recovers the annotation from the original resource when patching it and reads it to the desired Service Account to avoid removing the annotation and triggering OCP to create a new Secret. This is not a great solution, but until we have Server Side Apply support, this is the best we can do.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/kubernetes/ServiceAccountOperator.java [hunk 1]
```diff
@@ -42,6 +46,27 @@ protected Future<ReconcileResult<ServiceAccount>> internalUpdate(Reconciliation
...
+        // OpenShift 4.16 will create an image pull secret, add it to the image pull secrets list and attach an
+        // annotation openshift.io/internal-registry-pull-secret-ref to the Service Account with the name of the image
+        // pull secret. If Strimzi removes this annotation by patching the resource in the next reconciliation,
+        // OpenShift will create another Secret, add it to the image pull secret list and add back the annotation.
+        // This way, every reconciliation will for every Service Account make OCP generate new Secret and in a little
+        // while, the namespace is full of these Secrets.
+        //
+        // The code below recovers the annotation from the original resource when patching it and reads it to the
+        // desired Service Account to avid removing the annotation and triggering OCP to create new Secret. This is not
+        // a great solution, but until we have Server Side Apply support, this is the best we can do.
```
Comment (fvaleri) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/kubernetes/ServiceAccountOperator.java:58]:
```suggestion  
// The code below copies the pull secret ref annotation from the current service account to the desired service account (reconciliation result). 
// This workaround could be removed once we have Server Side Apply support.         
```

**Review by fvaleri — APPROVED** [review]
LGTM. Do we want to log an issue for getting rid of this workaround once the Server Side Apply is merged?

**Comment by k-wall** [comment]
I wanted to satisfy myself that this fix was sufficient (Is this the only SA annotation that needs special casing, should it be a pattern?)  The PR that made the change appears to be  https://github.com/openshift/openshift-controller-manager/pull/288/.

It looks to me like the change made is sufficient.  lgtm.

**Review by ppatierno — APPROVED** [review]

**Pull request merged**